add support for nested testsuites/testsuite reports

abendt opened the pull request:
lambdacd-junit assumes that the generated XML report contains a top-level `testsuite` element.
We use the jest-junit-reporter to generate reports for our JS tests. Its reports contains a top-level `testsuites` element which contains nested `testsuite` elements.

we adapted lambdacd-junit to support this report format.

thilo11 commented:
Thank you! It would be great if you could update the Readme as well to explain how this works in a Jest setup. There are also a few labels starting with an absolute path like '/Users/alphonse.bendt/...' in some test files. They should't contain user paths.

abendt commented:
I added a jest example. please check if this is what you had in mind.

The labels with absolute paths were generated by the jest junit reporter i used (https://github.com/michaelleeallen/jest-junit-reporter).

In the meantime I switched to a different reporter https://github.com/palmerj3/jest-junit. It's generated reports do not contain absolute path anymore. Also its better configurable

thilo11 commented:
thank you! I just merged it into master. I hope I get a new version out soon.
